src/component.jl

```diff
@@ -190,37 +243,115 @@ function get_time_series_timestamps(
     ))
 end
 
+"""
+Return a vector of timestamps from a cached Forecast instance.
+"""
 function get_time_series_timestamps(
     component::InfrastructureSystemsComponent,
-    time_series::TimeSeriesData,
+    forecast::Forecast,
+    start_time::Dates.DateTime;
+    len::Union{Nothing, Int} = nothing,
+)
+    return TimeSeries.timestamp(get_time_series_array(
+        component,
+        forecast,
+        start_time;
+        len = len,
+    ))
+end
+
+"""
+Return a vector of timestamps from a cached StaticTimeSeries instance.
+"""
+function get_time_series_timestamps(
+    component::InfrastructureSystemsComponent,
+    time_series::StaticTimeSeries,
+    start_time::Union{Nothing, Dates.DateTime} = nothing;
+    len::Union{Nothing, Int} = nothing,
 )
-    return TimeSeries.timestamp(get_time_series_array(component, time_series))
+    return TimeSeries.timestamp(get_time_series_array(
+        component,
+        time_series,
+        start_time;
+        len = len,
+    ))
 end
 
 """
-Return an Array of values for the requested time series parameters.
+Return an Array of values from storage for the requested time series parameters.
+
+If the data size is small and this will be called many times, consider using the version
+that accepts a cached TimeSeriesData instance.
 """
 function get_time_series_values(
     ::Type{T},
     component::InfrastructureSystemsComponent,
     name::AbstractString;
     start_time::Union{Nothing, Dates.DateTime} = nothing,
     len::Union{Nothing, Int} = nothing,
+    ignore_scaling_factors = false,
 ) where {T <: TimeSeriesData}
     return TimeSeries.values(get_time_series_array(
         T,
         component,
         name;
         start_time = start_time,
         len = len,
+        ignore_scaling_factors = ignore_scaling_factors,
+    ))
+end
+
+"""
+Return an Array of values for one forecast window from a cached Forecast instance.
+"""
+function get_time_series_values(
```

Comment:
@jd-lara Now you can call `get_time_series_array`, `get_time_series_values`, and `get_time_series_array` by passing a component and a cached time series object.  That should satisfy all your needs in PSI.

Reply:
This looks great